cli/run_util_test.go

```diff
@@ -515,3 +519,138 @@ workflows:
 	results, err := runWorkflowWithConfiguration(time.Now(), "target", config, []envmanModels.EnvironmentItemModel{})
 	require.Equal(t, 1, len(results.StepmanUpdates))
 }
+
+func TestInitializeStepDir(t *testing.T) {
+	tempTestResultsDir, err := pathutil.NormalizedOSTempDirPath("testing")
+	if err != nil {
+		t.Fatalf("failed to create testing dir, error: %s", err)
+	}
+
+	if err := os.Setenv("BITRISE_TEST_RESULTS_DIR", tempTestResultsDir); err != nil {
+		t.Fatalf("failed to set env, error: %s", err)
+	}
+
+	if os.Getenv("BITRISE_TEST_RESULT_DIR") != "" {
+		t.Fatal("BITRISE_TEST_RESULT_DIR should be empty")
+	}
+
+	var additionalEnvironments []envmanModels.EnvironmentItemModel
+
+	testDir, err := initializeStepDir(&additionalEnvironments)
+	if err != nil {
+		t.Fatalf("failed to create test dir, error: %s", err)
+	}
+
+	if filepath.Dir(testDir) != tempTestResultsDir {
+		t.Fatal("BITRISE_TEST_RESULT_DIR should be a child of BITRISE_TEST_RESULTS_DIR")
+	}
+
+	if len(additionalEnvironments) != 1 {
+		t.Fatal("should be only one additionalEnvironments")
+	} else {
+		key, value, err := additionalEnvironments[0].GetKeyValuePair()
+		if err != nil {
+			t.Fatalf("failed to get GetKeyValuePair, error: %s", err)
+		}
+		if key != "BITRISE_TEST_RESULT_DIR" {
+			t.Fatal("key should be BITRISE_TEST_RESULT_DIR")
+		}
+		if value != testDir {
+			t.Fatal("value should be the generated test dir path")
+		}
+	}
+
+	if exists, err := pathutil.IsDirExists(testDir); err != nil {
+		t.Fatalf("failed to check if dir exists, error: %s", err)
+	} else if !exists {
+		t.Fatal("BITRISE_TEST_RESULT_DIR path should exists on the FS")
+	}
+}
+
+func TestNormalizeStepDir(t *testing.T) {
+	t.Log("test empty dir")
+	{
+		testDirPath, err := pathutil.NormalizedOSTempDirPath("testing")
+		if err != nil {
+			t.Fatalf("failed to create testing dir, error: %s", err)
+		}
+
+		testResultStepInfo := models.TestResultStepInfo{}
+
+		exists, err := pathutil.IsDirExists(testDirPath)
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if !exists {
+			t.Fatal("test dir should exits")
+		}
+
+		if err := normalizeTestDir(testDirPath, testResultStepInfo); err != nil {
+			t.Fatalf("failed to normalize test dir, error: %s", err)
+		}
+
+		exists, err = pathutil.IsDirExists(testDirPath)
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if exists {
+			t.Fatal("test dir should not exits")
```

Comment:
The enclosing block can fail with `test dir should exits` above and with `test dir should not exits` here.

Consider how failure is output and write meaningful messages, e.g.: `test setup: could not create test dir` and `failed to clean up empty folder` respectively.

Reply:
Exactly, this check is after a removal, and I would like to se an error message that hey, this should not exists, you've just removed.